src/api/c/blas.cpp

```diff
@@ -127,7 +127,13 @@ af_err af_matmul(af_array *out,
         dim4 lDims = lhsInfo.dims();
         dim4 rDims = rhsInfo.dims();
 
+        //FIXME(pradeep) Move backend specific check to backend
```

Comment:
Are your plans to resolve this FIXME in this PR or in a future one?

Reply:
I think I will add the broadcast support for CPU and OpenCL as well if doesn't take too much effort.

Reply:
:+1: